sunpy/map/mapbase.py

```diff
@@ -1965,16 +1965,40 @@ def draw_rectangle(self, bottom_left, *, width: u.deg = None, height: u.deg = No
         axes.add_artist(rect)
         return [rect]
 
-    @u.quantity_input
-    def draw_contours(self, levels: u.percent, axes=None, **contour_args):
+    def _process_levels_arg(self, levels):
+        """
+        Accept a percentage or dimensionless or map unit input for contours.
+        """
+        if not hasattr(levels, 'unit'):
+            raise TypeError("The levels argument has no unit attribute, it should be an Astropy Quantity object.")
+
+        if levels.unit.is_equivalent(u.percent):
+            return 0.01 * levels.to('percent').value * np.nanmax(self.data)
+
+        elif self.unit is not None:
+            try:
+                return levels.to_value(self.unit)
+            # Catch cases where level can't be converted or isn't a Quantity
+            except (AttributeError, u.UnitConversionError) as e:
+                raise u.UnitsError(
+                    f'level must be an astropy quantity convertible to {self.unit}') from e
+        elif levels.unit.is_equivalent(u.one):
```

Comment:
@dstansby Without looking at your method I went for requiring a unit, rather than allowing a bare array. Thoughts?

Reply:
👍 , you'll need to check if any of of the existing contour changelogs need changing to reflect this